flecsi/topology/set_utils.h

```diff
@@ -121,21 +121,21 @@ struct find_set_index_space_u<0, TUPLE, ENTITY> {
   }
 };
 
-template<size_t INDEX, class TUPLE, class MAP_TYPE>
+template<size_t INDEX, class TUPLE, class FLECSI_MAP_TYPE>
```

Comment:
Wouldn't it make more sense to use the common `PascalCase` for template parameters rather than prefixing them like they need to be globally unique?  (And why is HPX defining a macro with as generic a name as `MAP_TYPE` anyway?)

Reply:
I completely agree on the suggested naming convention of using `CamelCase` for template arguments. Also `ALL_CAPS` should be used for macros only. This is however outside of our control. Both conventions have long been established in modern C++ libraries (starting with Boost).

The `MAP_TYPE` symbol is not defined in HPX but in some other C system header that is indirectly included by HPX. I have not checked where it actually is defined.

Reply:
Are you saying that the FleCSI template parameter names are out of your control?  If so, conveniently we have [rewritten the style guide](https://github.com/laristra/flecsi/blob/refactor/sphinx/src/developer-guide/style-guide.rst) and removed the all-caps template parameter rule.  (I note that even that new version probably needs some tweaking.)  So can we have `MapType` (or something even more informative) instead?

Meanwhile, it appears that `MAP_TYPE` is defined for `mmap` on (some?) Linux systems; I think it's silly that they added a non-ugly name for that bitmask, but it's not unreasonable to suppose that such headers would add additional `MAP_*` macros.

Reply:
I meant, that the file in question is not part of this PR (we only touched it to circumvent the symbol clash) and not part of our responsibility. I can certainly change the symbol to `MapType`, however.

Reply:
Well, I'm sorry you happened to be the ones to come across the incompatibility, but if anyone's going to change it it's got to be to something better than `FLECSI_MAP_TYPE` (which looks even more like a macro).  Thanks!